src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -95,12 +91,9 @@ public static AsyncVoidMethodBuilder Create()
         /// <summary>Completes the method builder successfully.</summary>
         public void SetResult()
         {
-            Task taskIfDebuggingEnabled = this.GetTaskIfDebuggingEnabled();
-            if (taskIfDebuggingEnabled != null)
+            if (AsyncCausalitySupport.LoggingOn)
             {
-                if (DebuggerSupport.LoggingOn)
-                    DebuggerSupport.TraceOperationCompletion(CausalityTraceLevel.Required, taskIfDebuggingEnabled, AsyncStatus.Completed);
-                DebuggerSupport.RemoveFromActiveTasks(taskIfDebuggingEnabled);
```

Comment:
This RemoveFromActiveTasks got lost. I will look into it.

Reply:
It seems that `RemoveFromActiveTasks` should be part of `Task`. It feels logical to remove it in the same component that added it. Hence it should be removed both here and in `LogExistingTaskCompletion`. Unfortunately neither CoreRT nor CoreCLR repositories don't have any history on this. I'd be happy for an additional confirmation.

Reply:
I agree that the tracing could be factored better. It is tricky to refactor it e.g. debugger has assumptions about various details of it.

For this PR, I would:
- Use CoreCLR version as source of truth
- Leave any bugfixing or refactoring for later

Reply:
Sounds reasonable.